velox/common/file/File.cpp

```diff
@@ -238,6 +246,21 @@ uint64_t LocalReadFile::preadv(
   return totalBytesRead;
 }
 
+folly::SemiFuture<uint64_t> LocalReadFile::preadvAsync(
+    uint64_t offset,
+    const std::vector<folly::Range<char*>>& buffers) const {
+  auto promise = std::make_unique<folly::Promise<uint64_t>>();
```

Comment:
nit no need to heap allocate the promise

auto [p, f] = folly::makePromiseContract<uint64_t>();

then move the promise into the lambda

Reply:
Also the compilation failed by `setTry` because `p` is const.

Reply:
you can add mutable to the lambda to the executor